internal/http/services/owncloud/ocdav/move.go

```diff
@@ -113,38 +113,18 @@ func (s *svc) handleSpacesMove(w http.ResponseWriter, r *http.Request, srcSpaceI
 	}
 
 	sublog := appctx.GetLogger(ctx).With().Str("spaceid", srcSpaceID).Str("path", r.URL.Path).Logger()
-	client, err := s.getClient()
-	if err != nil {
-		sublog.Error().Err(err).Msg("error getting grpc client")
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-
-	// retrieve a specific storage space
-	srcRef, status, err := spacelookup.LookUpStorageSpaceReference(ctx, client, srcSpaceID, r.URL.Path, true)
-	if err != nil {
-		sublog.Error().Err(err).Msg("error sending a grpc request")
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
 
-	if status.Code != rpc.Code_CODE_OK {
-		errors.HandleErrorStatus(&sublog, w, status)
+	srcRef := spacelookup.MakeStorageSpaceReference(srcSpaceID, r.URL.Path)
+	if srcRef == nil {
+		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
 	dstSpaceID, dstRelPath := router.ShiftPath(dst)
 
-	// retrieve a specific storage space
-	dstRef, status, err := spacelookup.LookUpStorageSpaceReference(ctx, client, dstSpaceID, dstRelPath, true)
-	if err != nil {
-		sublog.Error().Err(err).Msg("error sending a grpc request")
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-
-	if status.Code != rpc.Code_CODE_OK {
-		errors.HandleErrorStatus(&sublog, w, status)
+	dstRef := spacelookup.MakeStorageSpaceReference(dstSpaceID, dstRelPath)
+	if dstRef == nil {
```

Comment:
The reference will never be nil.

Reply:
> dstSpaceID is constructed from a header, net.ParseDestination cuts off the baseURI which might return an empty path. ShiftPath will then also return an empty path

Sure if you want to check this stuff then you have to do it explicitly in line 124 since `MakeStorageSpaceReference` will never validate anything and never return nil. Or you change `MakeStorageSpaceReference` to not ignore the error.